### DISCUSSION TITLE
Doris目前（master或者2.1之后）是否还可以支持jdk8?

### DISCUSSION THREAD

**Discussion by hongkunxu** [apache/doris#33688] [A - General / Q&A]
Hi @BePPPower 
我看到你上个月提的一个PR: https://github.com/apache/doris/commit/f73ea301534b78c99e203c323b784ebd8e259a27
这是一个强制使用jdk17的PR，我想了解的是这个是因为jdk17高性能所以推动大家使用jdk17的方式，还是因为trino spi等问题必须要运行的时候使用jdk17

**Comment by BePPPower** [comment]
Hi, INNOCENT-BOY, we need to upgrade to JDK 17 because we want to utilize Trino SPI.

**Reply by OxalisCu** [reply]
那请问不支持jdk8的话还能使用apache/doris:build-env-ldb-toolchain-latest这个源码镜像进行编译吗，这个镜像只支持jdk8和jdk11，不支持jdk17

**Reply by BePPPower** [reply]
Actually, the latest version of the `apache/doris:build-env-ldb-toolchain-latest` image contains both JDK 8 and JDK 17. If your `apache/doris:build-env-ldb-toolchain-latest` does not have JDK 17, you need to check if it is the latest version. You can check the update time in docker/README.md.

Reference: https://doris.apache.org/zh-CN/docs/dev/install/source-install/compilation-with-docker?_highlight=build&_highlight=env&_highlight=ldb&_highlight=toolchain&_highlight=latest#%E4%B8%8B%E8%BD%BD-doris-%E6%9E%84%E5%BB%BA%E9%95%9C%E5%83%8F

**Reply by OxalisCu** [reply]
确实支持，没仔细看，文档里面把 jdk17 写成了 jdk11
![image](https://github.com/apache/doris/assets/64067746/157ce67e-4721-4ce4-a2ca-d45020ccc086)

**Reply by BePPPower** [reply]
https://github.com/apache/doris-website/pull/658

**Reply by OxalisCu** [reply]
再请问一下，目前master最新代码需要jdk17作为依赖，在对 doris 进行编译调试过程中遇到下面问题

1. 开发者手册中 [Doris Docker 快速搭建开发环境](https://doris.apache.org/zh-CN/community/developer-guide/docker-dev) 里面使用了 apache/incubator-doris:build-env-latest 镜像最近更新是2年前，并且只支持 java11，对于master代码是否已经使用不了？
![image](https://github.com/apache/doris/assets/64067746/68d88c2a-c1f0-497c-8f25-196f1aa81d61)

2. 我目前使用 apache/doris:build-env-ldb-toolchain-latest 这个镜像作为编译环境编译成功，并希望在docker镜像里进行远程调试，但是发现其gdb缺少相关依赖，是否有新的可供调试的镜像，或者是该如何解决依赖问题

```bash
/var/local/ldb-toolchain/bin/gdb: error while loading shared libraries: libsource-highlight.so.4: cannot open shared object file: No such file or directory
```
```bash
/var/local/ldb-toolchain/bin/gdb: symbol lookup error: /var/local/ldb-toolchain/bin/gdb: undefined symbol: ZN9srchilite15SourceHighlight9highlightERSiRSoRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESA
```
